contract-tests/test_v3_api.py

```diff
@@ -0,0 +1,240 @@
+import json
+import jsonschema
+import magic
+import pytest
+
+from os.path import join, dirname
+from random import randint
+
+
+def _load_json_schema(filename):
+    relative_path = join("schemas", filename)
+    absolute_path = join(dirname(__file__), relative_path)
+
+    with open(absolute_path) as schema_file:
+        return json.loads(schema_file.read())
+
+
+def assert_valid_schema(data, schema_file):
+    schema = _load_json_schema(schema_file)
+    return jsonschema.validate(data, schema)
+
+
+def test_v3_actions(conf, requests_session):
+    # Get a random action and validate it
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("No list of v3 actions found")
+
+    key = randint(0, len(data["results"]) - 1)
+    action_id = data["results"][key]["id"]
+
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/{}/".format(action_id))
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("Could not find a v3 action with the ID {}".format(action_id))

```

Comment:
I think it should be an error for an action to be published in the list, but not to have a detail page.

Reply:
I added those in because tests kept failing on CI when they passed every single time on my own laptop when using https://normandy.cdn.mozilla.net as the server. 

I'm open to suggestions on how to solve this chicken-and-egg issue.

Reply:
The test skip for "there is no data in the listing" seem good. I think we should keep those. The second check of "there was data in the listing, but there is not detail" were the ones I had a problem with. It seems that you've already removed those though. I think we're good here.